youtube_dlc/extractor/youtube.py

```diff
@@ -1784,21 +1796,25 @@ def extract_embedded_config(embed_webpage, video_id):
                     # Try looking directly into the video webpage
                     ytplayer_config = self._get_ytplayer_config(video_id, video_webpage)
                     if ytplayer_config:
-                        args = ytplayer_config['args']
-                        if args.get('url_encoded_fmt_stream_map') or args.get('hlsvp'):
-                            # Convert to the same format returned by compat_parse_qs
-                            video_info = dict((k, [v]) for k, v in args.items())
-                            add_dash_mpd(video_info)
-                        # Rental video is not rented but preview is available (e.g.
-                        # https://www.youtube.com/watch?v=yYr8q0y5Jfg,
-                        # https://github.com/ytdl-org/youtube-dl/issues/10532)
-                        if not video_info and args.get('ypc_vid'):
-                            return self.url_result(
-                                args['ypc_vid'], YoutubeIE.ie_key(), video_id=args['ypc_vid'])
-                        if args.get('livestream') == '1' or args.get('live_playback') == 1:
-                            is_live = True
-                        if not player_response:
-                            player_response = extract_player_response(args.get('player_response'), video_id)
+                        args = ytplayer_config.get("args")
+                        if args is not None:
+                            if args.get('url_encoded_fmt_stream_map') or args.get('hlsvp'):
+                                # Convert to the same format returned by compat_parse_qs
+                                video_info = dict((k, [v]) for k, v in args.items())
+                                add_dash_mpd(video_info)
+                            # Rental video is not rented but preview is available (e.g.
+                            # https://www.youtube.com/watch?v=yYr8q0y5Jfg,
+                            # https://github.com/ytdl-org/youtube-dl/issues/10532)
+                            if not video_info and args.get('ypc_vid'):
+                                return self.url_result(
+                                    args['ypc_vid'], YoutubeIE.ie_key(), video_id=args['ypc_vid'])
+                            if args.get('livestream') == '1' or args.get('live_playback') == 1:
+                                is_live = True
+                            if not player_response:
+                                player_response = extract_player_response(args.get('player_response'), video_id)
```

Comment:
Is any of this stuff going to be relevant with the new player? Same with the captions stuff above. I don't know what this stuff is for but will more changes have to be made later to support this stuff in the new player?

Reply:
Good question. I am pretty confident that the automatic caption stuff works, all other code is from previous youtube player versions and basically does the same in a different way.

The stuff you marked seems to be about "Is this a live stream?", "Is this a previewable payed video?" and a third thing I do not understand (the "url_encoded_fmt_stream_map").

I checked a livestream, that part is fine. It is checked again from the video_details (which is extracted from the player_response) in line 1976. This even happens for the "old" player.
```
     if is_live is None:
            is_live = bool_or_none(video_details.get('isLive'))
```
For payed video with a preview, yt-dlc just says "This video requires payment to watch" (I used a movie as an example). I'm not quite sure when this code is used, and the example in the comment is down :/

Reply:
As there haven't been any more comments the last 3 days I assume the conversation is resolved.